python/qibuild/parallel_builder.py

```diff
@@ -151,6 +159,14 @@ def _find_job_by_name(self, name):
 
         return None
 
+    def _has_any_worker_failed(self):
+        for worker in self._workers:
+            with worker.result_lock:
+                if not worker.result.ok:
```

Comment:
Why not just check `if worker.result.failed_project`
That way you wouldn't need the lock because it would be an atomic operation